benchmarks/HOTPOTQA/generate.py

```diff
@@ -1,16 +1,30 @@
 import argparse
 import os
 import re
+import json
+import random
 
 from datasets import load_dataset
 from transformers import AutoModelForCausalLM, AutoTokenizer
 
 from rageval.models import OpenAILLM
-from prompts import (FEW_SHOT_EXAMPLES, PROMPT)
+from prompts import (DISTRACTOR_FEW_SHOT_EXAMPLES, FULLWIKI_FEW_SHOT_EXAMPLES, OPENDOMAIN_FEW_SHOT_EXAMPLES,
+                     DISTRACTOR_PROMPT, FULLWIKI_PROMPT, OPEN_DOMAIN_PROMPT)
 
 from typing import Tuple
 
 
+def reform_context(example):
+    new_context = []
+    for entry in example['context']:
+        doc = {}
+        doc['title'] = entry[0]
+        doc['sentences'] = entry[1]
+        new_context.append(doc)
+    example['context'] = new_context
+    return example
+
+
 def process_response(response: str) -> Tuple[str, str]:
```

Comment:
The current encoding method for `gt_sent_ids` and `supporting_answer` is too redundant. Consider using a tuple in the format `(title, sent_id)` or other methods as the new index for each `supporting_answer` to simplify it. Note that you may need to adjust the F1 score implementation, as it currently only accepts string input and automatically normalizes the text.